src/app/BreedProcessor.ts

```diff
@@ -16,29 +16,34 @@ export class BreedProcessor {
     this.randomResultInterfaceFox = new FirstDice();
   }
 
-  processBreedPhase({ theHerd }: Player): void {
-    const wolf = this.getRandomResult(
-      this.randomResultInterfaceWolf.getRandomValue(),
-    );
-    const fox = this.getRandomResult(
-      this.randomResultInterfaceFox.getRandomValue(),
-    );
-    if (typeof fox !== 'object') {
-      if (typeof wolf !== 'object') {
-        this.breedAnimals(fox, wolf, theHerd);
+  processBreedPhase({ theHerd }: Player): [AnimalNames, AnimalNames] {
+    const wolfDiceResult = this.randomResultInterfaceWolf.getRandomValue();
+    const foxDiceResult = this.randomResultInterfaceFox.getRandomValue();
+    if (foxDiceResult !== AnimalNames.FOX) {
+      if (wolfDiceResult !== AnimalNames.WOLF) {
+        this.breedAnimals(foxDiceResult, wolfDiceResult, theHerd);
```

Comment:
Great, now it's much clearer! I was wondering though, is there any possibility these animals will change?

Reply:
it depends what client would like to have, for basic version this will do